python/shipDigiReco.py

```
@@ -75,43 +83,93 @@ def __init__(self, finput, fout, fgeo) -> None:
+            iHit += [station, layer, view, straw, trID, x, y, z, time, px, py, pz, 0, 0, 0, 0, deltaE]
+            self.strawHits.push_back(iHit)
+
+        candidates = []
+
+        if global_variables.patRec == "AR":  # Or whatever flag you use
+            # PatRec Source
+            raw_candidates = self.findTracks()
+            for cand in raw_candidates:
+                # cand is already in our dictionary format from the previous step
+                candidates.append(cand)
+        elif globa_variables.patRec == "Truth":
```

Ruff (0.15.12)

[error] Undefined name `globa_variables`

(F821)